javascript/ql/src/semmle/javascript/security/dataflow/ExceptionXss.qll

```diff
@@ -24,16 +25,31 @@ module ExceptionXss {
     node = DataFlow::globalVarRef("console").getAMemberCall(_).getAnArgument()
   }
 
+  /**
+   * Holds if `t` is `null` or `undefined`.
+   */
+  private predicate isNullOrUndefined(InferredType t) {
+    t = TTNull() or
+    t = TTUndefined()
+  }
+
   /**
    * Holds if `node` can possibly cause an exception containing sensitive information to be thrown.
    */
   predicate canThrowSensitiveInformation(DataFlow::Node node) {
-    not isUnlikelyToThrowSensitiveInformation(node) and 
+    not isUnlikelyToThrowSensitiveInformation(node) and
     (
       // in the case of reflective calls the below ensures that both InvokeNodes have no known callee.
-      forex(DataFlow::InvokeNode call | node = call.getAnArgument() | not exists(call.getACallee()))
+      forex(DataFlow::InvokeNode call | call = getEnclosingCallNode(node) |
+        not exists(call.getACallee())
+      )
       or
       node.asExpr().getEnclosingStmt() instanceof ThrowStmt
+      or
+      exists(DataFlow::PropRef prop |
+        node.getEnclosingExpr() = prop.getPropertyNameExpr() and
```

Comment:
Do we really need to step back to the `Expr` layer on both sides of this equality? Couldn't both sides remain in the `DataFlow` layer?

Reply:
I would have liked to, but the only relevant predicate on `PropRef` is `getPropertyNameExpr()`.  
I can change it to `node = DataFlow::valueNode(prop.getPropertyNameExpr())`?

Reply:
Oh, I see. I thought we had `PropRef::getPropertyNameNode` or something like that. Lets pick your proposed change, and postpone a core library change for now.

(I think the `getEnclosingExpr` originally threw me of due to the PR also containing `getEnclosingCall`, and then it snowballed from there)